frontend/src/store/modules/maps.js

```diff
@@ -52,9 +53,15 @@ const getters = {
 };
 
 const actions = {
+
   async getMapsListing({ commit }, model) {
     const payload = { model: model.apiName, version: model.apiVersion };
     const mapsListing = await mapsApi.fetchMapsListing(payload);
+    let avail2D = false;
+    Object.keys(mapsListing).forEach((cat) => {
+      avail2D = mapsListing[cat].some(x => x.svgs.length === 1) || avail2D;
```

Comment:
I wonder if it makes sense to have `x.svgs.length > 0` since Cytosol has multiple svg:s. But I don't think any model has only Cytosol as 2d map so this should also be fine

Reply:
I think this is a good suggestion. I'm also wondering if we can short-circuit the iteration so we don't have to go through every item. Maybe something like the following. 

```javascript
    // eslint-disable-next-line
    for (const { svgs } of Object.values(mapsListing).flat()) {
      if (svgs.length > 0) {
        avail2D = true;
        break;
      }
    }
```

Please note that I'm not actually sure if this is faster.. both ways are probably fast enough to have any noticeable effect but this makes more sense to me logically.

Reply:
Thanks for the suggestions :)

I more or less copied the `svgs.length === 1` from `MapListings.vue` and didn't dare to change it. But yes, `> 0` sounds better!

I also think that your idea makes much sense @e0. Do you prefer the `for` loop, even though `eslint` gets upset about it?

Reply:
Updated to the suggestion above in 26a2acc62d9759498d19a333708c528a05ed8baa

Reply:
Sorry for the slow response @MalinAhlberg. Good that you thought about the `eslint` warning. It's not ideal but I looked up how to [short-circuit a `forEach`](https://stackoverflow.com/a/2641374) and thought this was the better alternative.

There is also another variant using `every` (it immediately returns when a falsy value is returned in the iterator but I still think the `for` loop is more readable.

```javascript
    Object.values(mapsListing).flat().every(({ svgs }) => {
      if (svgs.length > 0) {
        avail2D = true;
        return false;
      }
      return true;
    });
```

Reply:
Thanks for the link and examples!! Using `every` is tempting, I had to try out the one-line-version. But it got very unreadable, especially with the negated truth value checking :smile:
```js
    avail2D = ! Object.values(mapsListing).flat().every(function(map) {return map.svgs.length === 0});
```

Reply:
That looks pretty good I think! I guess you can make this slightly shorter with using an arrow function and knowing that `0` is a falsy value in JavaScript.

```javascript
    avail2D = !Object.values(mapsListing).flat().every((m) => m.svgs.length);
```

Reply:
Aha, nice :star: !!! Maybe not very readable for random humans, but indeed nice.

> Edit: On second thought, this might not work as intended. If the first compartment map does not have any SVGs but the other maps have, then this gives the wrong result right?

Since it flattened, that shouldn't be a problem?

Reply:
I meant my follow up suggestion, since it only looks at the length and doesn't check if it is equal to 0, it would be the inverse or your original one-liner.

Reply:
Ah, I see, good point. So `m.svgs` would be null(-ish) and hence falsy?
But then it would have to be like this
```js
    avail2D = !Object.values(mapsListing).flat().every((m) => m.svgs.length === 0);

```
Still nice!

Reply:
Random idea: instead of passing on this computation to the client browser, would it make sense to have it in the API response?

Reply:
I think it's a very reasonable idea, but I'm not sure it's worth the refactoring? It's not a heavy computation. But I'm up for changing if there is an agreement that we should do so :)

Reply:
> Random idea: instead of passing on this computation to the client browser, would it make sense to have it in the API response?

Interesting suggestion but I would not do this for the following reasons.

1. The performance difference is near-zero considering all the other intensive things that the frontend must handle such as rendering the maps.
2. This serves a very frontend specific purpose. IMO the data returned from the backend should be as flexible as possible so that the data is cleaner for API users.